ori/volume/server/server.go

```diff
@@ -126,52 +123,39 @@ var _ ori.VolumeRuntimeServer = (*Server)(nil)
 //+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=storage.api.onmetal.de,resources=volumes,verbs=get;list;watch;create;update;patch;delete
 
-func New(cfg *rest.Config, opts Options) (*Server, error) {
-	setOptionsDefaults(&opts)
-
-	c, err := client.New(cfg, client.Options{
-		Scheme: scheme,
-	})
+func New(opts Options, provisioner Provisioner) (*Server, error) {
+	classes, err := ReadVolumeClasses(opts.PathAvailableVolumeClasses)
 	if err != nil {
-		return nil, fmt.Errorf("error creating client: %w", err)
+		return nil, fmt.Errorf("unable to get volume classes: %w", err)
 	}
 
-	return &Server{
-		client: c,
-		idGen:  opts.IDGen,
-
-		namespace: opts.Namespace,
-
-		rookNamespace:            opts.RookNamespace,
-		rookClusterName:          opts.RookClusterName,
-		rookPoolName:             opts.RookPoolName,
-		rookClientName:           opts.RookClientName,
-		rookPoolSecretName:       opts.RookPoolSecretName,
-		rookPoolMonitorConfigmap: opts.RookPoolMonitorConfigmap,
-		rookPoolStorageClass:     opts.RookPoolStorageClass,
-
-		driver:    opts.Driver,
-		wwnPrefix: opts.WwnPrefix,
+	setOptionsDefaults(&opts)
 
-		volumeClassSelector: opts.VolumeClassSelector,
+	return &Server{
+		idGen:                  opts.IDGen,
+		VolumeNameLabelName:    opts.VolumeNameLabelName,
+		AvailableVolumeClasses: classes,
+		provisioner:            provisioner,
+		inProgress:             map[string]sets.Empty{},
+		inProgressLock:         sync.Mutex{},
 	}, nil
 }
 
-func (s *Server) getManagedAndCreated(ctx context.Context, name string, obj client.Object) error {
-	key := client.ObjectKey{Namespace: s.namespace, Name: name}
-	if err := s.client.Get(ctx, key, obj); err != nil {
-		return err
+func ReadVolumeClasses(path string) (map[string]ori.VolumeClass, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read volume class json file (%s): %w", path, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error reading config file at %q: %w", path, err)
```
Using `%q` quotes the string. We're also not sure whether it's an actual JSON file at this point.